src/components/Footer.tsx

```diff
@@ -0,0 +1,79 @@
+import { FilterBy } from '../types/FilterBy';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  todos: Todo[];
+  filter: FilterBy;
+  onFilter: (selectedFilter: FilterBy) => void;
+  onDelete: (id: number) => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filter,
+  onFilter,
+  onDelete,
+}) => {
+  const todosCounter = todos.filter(todo => !todo.completed).length;
+  const todosCompleted = todos.some(todo => todo.completed);
+
+  const handleDeleteCompletedTodos = () => {
+    const allTodosCompleted = todos.filter(todo => todo.completed);
+
+    allTodosCompleted.map(todo => onDelete(todo.id));
+  };
+
+  return (
+    <>
+      <footer className="todoapp__footer" data-cy="Footer">
+        <span className="todo-count" data-cy="TodosCounter">
+          {todosCounter} items left
```

Comment:
Let's minimize array methods, as they can be heavy

```suggestion
  const allTodosCompleted = todos.filter(todo => todo.completed);

  const handleDeleteCompletedTodos = () => {
    allTodosCompleted.forEach(todo => onDelete(todo.id));
  };

  return (
    <>
      <footer className="todoapp__footer" data-cy="Footer">
        <span className="todo-count" data-cy="TodosCounter">
          {todos.length - allTodosCompleted.length} items left
```